Add image module

tcharding opened the pull request:
Add an `image` module, implement `testcontainers::Image` on a `Monero` struct.
The idea is that this module can be upstreamed to `testcontainers` once its
stable.

Commit:
Add image module

Add an `image` module, implement `testcontainers::Image` on a `Monero` struct.
The idea is that this module can be upstreamed to `testcontainers` once its
stable.

Commit:
Move Monero code blocks to top of file

The `Monero` struct is the main thing in this module, put all its code blocks at
the top and the args stuff at the bottom.

Commit:
Don't print args

We can get the args from docker, no need to print them.

	docker ps --no-trunc

Commit:
Un-comment commented out test

We want to run this test now.

Commit:
Use offline mode

The tests are currently hanging because we never get the log message we are
waiting for, monerod is trying to sync with the network. Use `--offline` so
there is no sync.

thomaseizinger commented on src/image.rs:
```diff
@@ -0,0 +1,271 @@
+    pub log_level: u32,
+}
+
+/// Sane defaults for a mainnet regtest instance.
```
I am curious, what is a "mainnet regtest instance"?

tcharding replied:
I'm confused about all this too. Seems for monero their networks also imply what branc/commit they are working off. `mainnet` and `stagenet` use release tagged commits. `testnet` uses tip of master branch. And, IIUC, `--regtest` (or regtest mode) uses same commit as mainnet and the same default settings i.e., port numbers but makes available some extra RPC commands (`generateblocks`) and allows one to set the mining difficulty with `--fixed-difficulty`. 

I will try to word this code comment better.

bonomat commented on src/image.rs:
```diff
@@ -0,0 +1,275 @@
+    Image,
+};
+
+// TODO: Switch to using 18081 port since --regtest is mainnet
```
I disagree here. `regtest` should use a different port than mainnet. While `regtest` uses `mainnet` settings `regtest` != `mainnet`. Imagine you run 2 nodes: `regtest` (for testing) and `mainnet` (for production).

tcharding replied:
Posting here for continuity, we chatted in a call. Going to use 48081 (since currently there is significance to the following port numbers within the monero community

18081 - mainnet
28081 - stagenet
38081 - testnet

thomaseizinger replied:
Does monero run on different ports per network by default?

This is being run as a docker container through testcontainers, so the port that is exposed on the host machine is some random one anyway and you have to fetch it with `get_mapped_port`.

More generally: While I get the argument, I am not convinced that it actually adds value. On the contrary, writing software against an Ethereum node (which always uses `8545`) is so much easier than for Bitcoin. Just have a look at our configuration file handling and all the extra code we have to write just to get the port settings correct.

If "I want to run two different nodes on my machine" is the only argument _for_ this behaviour, then I don't think it was a good trade-off by the Bitcoin team to implement this behaviour. The default case is to only run a single instance. I should not need to write a lot of code to handle the default case. If someone is running two at the same time, they should have to do some work to make that happen.

Commit:
Define consts for port numbers

The following port numbers have significance in the monero community

18081 - mainnet
28081 - stagenet
38081 - testnet

Lets define 48081 for regtest mode. Add consts to do so.
